File: Data Structures & Algorithms/permutation-string/submission-0.py
class Solution:
    def checkInclusion(self, s1: str, s2: str) -> bool:
        if len(s1) > len(s2):
            return False
        
        arr = list(s2)
        sorted_s1 = sorted(s1)
        # print(arr)

        for i in range(len(s2) - len(s1) + 1):
            sub_str = ''.join(arr[i : i + len(s1)])
            # print(sub_str, "\n")
            if sorted(sub_str) == sorted_s1:
                return True
        return False

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.cpp
class Solution {
public:
    int minCostClimbingStairs(vector<int>& cost) {
        int n = cost.size();
        vector<int> arr(n + 1);
        arr[0] = cost[0];
        arr[1] = cost[1];
        arr[n] = 0;
        for (int i = 2; i <= n; i++) {
            if (i == n) {
                arr[i] = min(arr[i - 2], arr[i - 1]);
            } else {
                arr[i] = min(arr[i - 2], arr[i - 1]) + cost[i];
            }
        }  
        return arr[n];
    }
};
